src/core/sock/sockinfo_tcp.cpp

```diff
@@ -1112,6 +1102,10 @@ ssize_t sockinfo_tcp::tcp_tx(xlio_tx_call_attr_t &tx_arg)
                 return tcp_tx_handle_sndbuf_unavailable(total_tx, is_dummy, is_non_file_zerocopy,
                                                         errno_tmp);
             }
+            if (tx_arg.xlio_flags & TX_FLAG_NO_PARTIAL_WRITE) {
+                // TODO Fix type overflow if user sends over UINT32_MAX bytes
```

Comment:
Would it be acceptable to return error for now if uint32 exceeded?

Reply:
as last resort yes. Other option is to make send buffer 64bit which will increase socket size by 4 bytes.

(not done yet)